lib/vet/tarball.js

```diff
@@ -9,16 +9,24 @@ module.exports = function(module, dir, url, json) {
   var moduleDir = path.join(dir, 'node_modules', module);
   return new Promise(function(resolve, reject) {
     var children = [];
-    var stream = got.stream(url).pipe(zlib.Unzip()).pipe(tar.Parse());
+    var stream = got.stream(url).pipe(zlib.Unzip()).pipe(new tar.Parse());
     stream.on('entry', function(arg) {
-      if (arg.type !== 'File') return;
+      if (arg.type !== 'File') { arg.resume(); return; }
       var parts = arg.path.split('/');
-      if (arg.path.indexOf('node_modules') !== -1) return;
-      if (parts[1] === 'package.json') return children.push(vetPackageJSON(moduleDir, arg));
-      if (parts[parts.length-1] === '.gitignore') return;
-      if (parts[parts.length-1].slice(-3) !== '.js') return;
+      if (arg.path.indexOf('node_modules') !== -1) { arg.resume(); return; }
+      if (parts[1] === 'package.json') {
+        return children.push(vetPackageJSON(moduleDir, arg).catch((err) => {
+          if (err.code === 'ENOENT' || err.message === 'Files dont match') return reject(new Error('Content of '+module+' does not match content of '+url));
```

Comment:
what is the error message in case the code is `ENOENT`? Do you think its worth it to log the actual message in the exception message to?

Reply:
One ENOENT error I found is thrown in getLocalFile where you are trying to get a local version of a particular file from what you are diffing against and the file does not exist locally. I think the ultimate result is what the exception message says, the content does not match, so I feel like the exception message is fine.